### PR TITLE
test(cover): render a pack-shaped template through core

### LINKED ISSUE #4092: test(cover): no suite renders a pack-shaped template through core, so the contract can drift silently
### Code of Conduct

- [x] I agree to follow the Code of Conduct

### Existing issues

- [x] I searched existing issues and this hasn't been reported yet

### What happened?

Nothing renders a pack-shaped template through core, so the published pack contract and the renderer can drift apart with every test still green.

Measured on `main`:

- `tests/template-packs.test.mjs` (382 lines) imports only `listTemplates` and `resolveTemplate`. Its three references to `build-cv-html.mjs` and `generate-cover-letter.mjs` are all in comments. It proves a pack is *found*, never that it *renders*.
- `validateTemplate` requires exactly 3 placeholders for `kind=cover`: `NAME`, `ROLE_TITLE`, `OPENING`. The pack cover contract defines 15.
- `RECIPIENT_BLOCK` and `DATELINE` appear **0 times** anywhere under `tests/`. (Positive control: `DATELINE` appears in `generate-cover-letter.mjs`, so the search works.)
- The suites that do call `buildHtml` build their own inline fixture templates, which naturally contain only the slots that test needs. None uses the contract's slot set.

So a template author gets a green `validateTemplate`, and the first thing that tells them the template is unrenderable is the render failing in front of a real application.

### Why this is the root cause, not a separate nicety

Both #4042 and #4068 are instances of it, found by rendering a real letter rather than by CI:

- #4042: `{{RECIPIENT_BLOCK}}` is required by the contract and was never filled by core, so every pack cover template failed at substitution.
- #4068: `{{DATELINE}}` carries the date under the contract, while core joined company and city into it, so a pack letter prints the company twice.

Neither was catchable by the existing tests, and neither would be catchable tomorrow if a 16th slot is added to the contract.

### Expected behavior

One suite that builds a template carrying the contract's full slot set, renders it through `buildHtml` with a fully populated payload, and asserts every slot is substituted and every supplied value reaches the output. That is the check that fails the moment the contract gains a slot core does not fill, or core starts filling one with the wrong thing.

The same argument applies to the CV side and `build-cv-html.mjs`, where #3775 was the same class of failure. I would rather land the cover half first and see whether the shape is right before proposing the CV half.

### Note on sequencing

Written against `main` today, this test fails, because `{{RECIPIENT_BLOCK}}` is genuinely unfilled until #4043 lands. I will stack the PR on #4069 so the suite is green when it arrives and guards both fixes from then on.

### CLI tool

Claude Code

### OS and Node.js version

macOS 26.6.2 / Node 26.8.1

---

I have a fix in progress and will open a PR shortly.

### PR DESCRIPTION
**Pull request by rubicon**
## What does this PR do?

Adds `tests/cover-pack-contract.test.mjs`, which renders a template carrying every slot the pack cover-letter contract declares and asserts core fills all of them. Nothing did this before, which is how the contract and the renderer drifted apart twice without a red test.

The gap is structural, not an oversight in any one suite. `validateTemplate` demands 3 placeholders for `kind=cover` while the contract declares 14. `tests/template-packs.test.mjs` proves a pack *resolves* but never renders one — its three references to the renderers are all comments. And every other cover suite builds its own inline fixture carrying only the slots that test needs, so none of them can see a slot core forgot.

The slot list in the test is a literal transcribed from the authoring docs rather than something derived from `generate-cover-letter.mjs`. A list read out of the renderer would agree with itself no matter what the contract says — which is exactly the check that was missing.

## Stacking

This branch sits on top of #4043 and #4069, so the diff below carries their commits too. Only the last one is new here:

| Commit | PR | New in this PR |
|---|---|---|
| `8079ed40` fill `{{RECIPIENT_BLOCK}}` | #4043 | no |
| `9fe49937` whitespace-only recipient field | #4043 | no |
| `bccab47c` gate `{{DATELINE}}` | #4069 | no |
| `a7bc35fb` pack-contract test | this PR | **yes** |

Review just `a7bc35fb`, or merge #4043 and #4069 first and this shrinks to the one file. Happy to rebase onto whatever lands.

## Related issue

Closes #4092

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Documentation / translation
- [x] Refactor (no behavior change)

Test-only — no source file changes in `a7bc35fb`.

## Verification

Run against the unfixed renderer as a positive control: all four tests fail on `main` and pass on this branch. Without that, a test over a contract the renderer already satisfies proves nothing.

```
# with main's generate-cover-letter.mjs
ℹ tests 4 / pass 0 / fail 4

# on this branch
ℹ tests 4 / pass 4 / fail 0
```

`node test-all.mjs`: 8693 passed, 0 failed.

## Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/career-ops-hq/career-ops/blob/main/CONTRIBUTING.md)
- [x] If this is a new feature or architecture change, I opened an issue first (bug fixes, providers, docs & translations are exempt — send those straight in)
- [x] My PR does not include personal data (CV, email, real names, scan results, or pipeline data)
- [x] I ran `node test-all.mjs` and all tests pass
- [x] My changes respect the [Data Contract](https://github.com/career-ops-hq/career-ops/blob/main/DATA_CONTRACT.md) (no modifications to user-layer files)
- [x] My changes align with the [project roadmap](https://github.com/career-ops-hq/career-ops/discussions/156)

## Summary

Adds cover-letter contract tests and fixes recipient and dateline rendering.

### User impact

Users can render pack-shaped cover-letter templates through `buildHtml` with all contract slots populated (`tests/cover-pack-contract.test.mjs:28-34`).

The renderer now:

- Fills `{{RECIPIENT_BLOCK}}` with escaped recipient lines (`generate-cover-letter.mjs:143-176`).
- Renders only the date in `{{DATELINE}}` when the recipient block has content (`generate-cover-letter.mjs:126-140`).
- Preserves the existing company, city, and date format when no recipient block renders (`tests/cover-dateline.test.mjs:61-80`).

The tests verify contract coverage, slot-specific contact and credential values, unresolved tokens, recipient edge cases, and HTML escaping (`tests/cover-pack-contract.test.mjs:79-124`; `tests/cover-recipient-block.test.mjs:39-110`).

### Files changed

- `generate-cover-letter.mjs`: Adds recipient-block rendering and conditional dateline behavior.
- `tests/cover-pack-contract.test.mjs`: Covers every documented cover-letter contract slot.
- `tests/cover-dateline.test.mjs`: Covers dateline behavior.
- `tests/cover-recipient-block.test.mjs`: Covers recipient rendering, escaping, and empty values.

No changes were made to `AGENTS.md`, `modes/`, `update-system.mjs`, `DATA_CONTRACT.md`, `providers/`, or `.github/`.

### CODE REVIEW

**Review by Scott-Emberson — COMMENTED** [review]
Test-axis only. The substance also touches `generate-cover-letter.mjs` (the dateline gating and recipient block, core), the maintainer's call; I am reviewing the owned tests.

The new `tests/cover-pack-contract.test.mjs` is sound and closes a real gap. It is the first test to drive a pack-shaped template with all fourteen contract slots through the real buildHtml, and it pins three things: the template renders without throwing, every one of the fourteen slots is substituted with no leftover token, and every payload value actually reaches the letter (the filled-but-empty class of bug). Its strength is that the slot list is a hand-transcribed literal from the pack authoring docs rather than derived from the renderer, so it detects contract-versus-renderer drift instead of agreeing with itself, which is exactly the gap where validateTemplate only demands three placeholders while the contract declares fourteen. I mutation-checked it: dropping a slot only this test covers reddens it alone, leaving an unresolved token reddens all three cover suites through the shared path, and resolving the recipient block to an empty div reddens only the every-value-reaches-the-letter assertion (so it asserts content, not just substitution). The sibling dateline and recipient-block tests are unchanged and still pass.

Owned tests sound; branch behind main. SIGNAL (generate-cover-letter core), so not surfacing as ready-to-approve; routing and merge are the maintainer's.

File: tests/cover-pack-contract.test.mjs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+test('every value the payload supplies reaches the letter', () => {
+  // Substitution alone is not enough: {{ACHIEVEMENTS_BLOCK}} was "filled" with
+  // empty <li> elements for the whole life of the achievements-shape bug.
+  const html = buildHtml(FULL, packTemplate());
+
+  for (const v of ['A Candidate', 'Head of Marketing', 'Jane Reviewer', 'Director of Talent',
+                   '100 Example Street', 'OPENING-TEXT', 'PROFILE-INTRO-TEXT', 'ACH-LEAD',
+                   'ACH-IMPACT', 'PROBLEMS-TEXT', 'CLOSING-TEXT', 'LANGUAGE-CLOSING-TEXT',
+                   'CRED-ONE', 'FOOTNOTE-ONE', 'September 11, 2026']) {
+    assert.ok(html.includes(v), `the payload supplied "${v}" and the letter does not carry it`);
```
[the remark this answers is not in the document]
Comment (rubicon) [reply tests/cover-pack-contract.test.mjs]: Confirmed and fixed in 93b8e424.

The finding held for the first loop. Every value now has a distinct sentinel and is asserted inside its own data-slot, via one slot/value table, never against the whole document.

The fixture also reused one string for two fields (letter.company / recipient.company, candidate.location / letter.city), so those covered neither side. De-colliding them let me add a negative assertion that DATELINE no longer repeats the company.

The slot wrapper moved from <div> to <section>. RECIPIENT_BLOCK emits nested <div>s that a non-greedy <div> regex closes at the first inner tag, truncating the slot it is meant to isolate.

Proof the change discriminates. Dropping r.name from buildRecipientBlock left all 4 tests GREEN before this commit, and REDDENS now. The greeting carried "Jane Reviewer", and the global includes() accepted it. Emptying CONTACT_LINE and re-joining company into the dateline each redden exactly one test too.

CONTACT_LINE/CREDENTIALS_BLOCK were already slot-scoped by 5d97cb1a, so that half of the finding was addressed before this pass.
